### PR TITLE
8290966: G1: Record number of PLAB filled and number of direct allocations

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  for evaluation in [JDK-8288966](https://bugs.openjdk.org/browse/JDK-8288966) I added statistics output that show the amount of PLAB fills and direct allocations; I think this is useful for similar evaluations in the future, so I kept and split it out from that change. Adds these values to the existing JFR event too.

Testing: PLAB related tests, gha

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290966](https://bugs.openjdk.org/browse/JDK-8290966): G1: Record number of PLAB filled and number of direct allocations

### Reviewers
 * [Sangheon Kim](https://openjdk.org/census#sangheki) (@sangheon - **Reviewer**) ⚠️ Review applies to [7b242dd1](https://git.openjdk.org/jdk/pull/9626/files/7b242dd1f5add285f68d8ce3b5174285fe50c176)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [7b242dd1](https://git.openjdk.org/jdk/pull/9626/files/7b242dd1f5add285f68d8ce3b5174285fe50c176)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9626/head:pull/9626` \
`$ git checkout pull/9626`

Update a local copy of the PR: \
`$ git checkout pull/9626` \
`$ git pull https://git.openjdk.org/jdk pull/9626/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9626`

View PR using the GUI difftool: \
`$ git pr show -t 9626`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9626.diff">https://git.openjdk.org/jdk/pull/9626.diff</a>

</details>

### CODE REVIEW

**Review by sangheon — APPROVED** [review]
Lgtm.

File: src/hotspot/share/gc/shared/gcHeapSummary.hpp [hunk 1]
```diff
@@ -188,11 +190,12 @@ class G1EvacSummary : public StackObj {
...
 public:
   G1EvacSummary(size_t allocated, size_t wasted, size_t undo_wasted, size_t unused,
-    size_t used, size_t region_end_waste, uint regions_filled, size_t direct_allocated,
-    size_t failure_used, size_t failure_waste) :
+    size_t used, size_t region_end_waste, uint regions_filled, size_t num_plab_filled,
+    size_t direct_allocated, size_t num_direct_allocated, size_t failure_used, size_t failure_waste) :
     _allocated(allocated), _wasted(wasted), _undo_wasted(undo_wasted), _unused(unused),
     _used(used),  _region_end_waste(region_end_waste), _regions_filled(regions_filled),
-    _direct_allocated(direct_allocated), _failure_used(failure_used), _failure_waste(failure_waste)
+    _num_plab_filled(num_plab_filled), _direct_allocated(direct_allocated), _num_direct_allocated(num_direct_allocated),
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/gcHeapSummary.hpp:197]: Maybe another line break in this line, to keep it's length a bit more reasonable?

File: src/hotspot/share/gc/shared/gcHeapSummary.hpp [hunk 2]
```diff
@@ -188,11 +190,12 @@ class G1EvacSummary : public StackObj {
   size_t _failure_waste;
 public:
   G1EvacSummary(size_t allocated, size_t wasted, size_t undo_wasted, size_t unused,
-    size_t used, size_t region_end_waste, uint regions_filled, size_t direct_allocated,
-    size_t failure_used, size_t failure_waste) :
+    size_t used, size_t region_end_waste, uint regions_filled, size_t num_plab_filled,
+    size_t direct_allocated, size_t num_direct_allocated, size_t failure_used, size_t failure_waste) :
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/gcHeapSummary.hpp:194]: [pre-existing] Shouldn't these parameters be indented to the parameter list start?  As it is, it's hard to spot where the parameters end and the initializers start.

**Review by kimbarrett — APPROVED** [review]
Looks good.  Just a couple whitespace comments.

**Comment by tschatzl** [comment]
Thanks @kimbarrett @sangheon for your reviews
/integrate

**Pull request closed without merging**